src/sentry/integrations/issues.py

```diff
@@ -150,15 +147,31 @@ def sync_status_outbound(self, external_issue, is_resolved, project_id, **kwargs
         """
         raise NotImplementedError
 
-    def get_external_project_id(self, data):
+    def should_unresolve(self, data):
         """
-        Given webhook data, pull out external project id
+        Given webhook data, check whether the status
+        category changed FROM "done" to something else,
+        meaning the sentry issue should be marked as
+        unresolved
+
+        >>> def should_unresolve(self, data):
+        >>>     client = self.get_client()
+        >>>     statuses = client.get_statuses()
+        >>>     done_statuses = [s['id'] for s in statuses if s['category'] == 'done']
+        >>>     return data['from_status'] in done_statuses \
+        >>>         and data['to_status'] not in done_statuses
```

Comment:
i realize these methods might require a lot of the same logic, but i'd rather make them general and easy to read/understand like this and then the specific integrations can make this DRYer by doing something like:

```python
def _get_done_statuses(self):
    client = self.get_client()
    statuses = client.get_statuses()
    return [s['id'] for s in statuses if s['category'] == 'done']

def should_unresolve(self, data):
    done_statuses = self.__get_done_statuses()
    return data['from_status'] in done_statuses \
        and data['to_status'] not in done_statuses

def should_resolve(self, data):
    done_statuses = self.__get_done_statuses()
    return data['to_status'] in done_statuses \
        and data['from_status'] not in done_statuses
```

^ @lauryndbrown i know you were concerned about that in a previous pr

Reply:
That's fine. Thanks for the example usage. I was just confused about how it would work.